st2actions/st2actions/runners/actionchainrunner.py

```diff
@@ -152,23 +154,51 @@ def run(self, action_parameters):
         fail = True
 
         while action_node:
+            fail = False
             error = None
+            resolved_params = None
             liveaction = None
-            fail = False
+
             created_at = datetime.datetime.now()
 
             try:
                 resolved_params = ActionChainRunner._resolve_params(
                     action_node=action_node, original_parameters=action_parameters,
                     results=context_result, chain_vars=self.chain_holder.vars)
+
                 liveaction = ActionChainRunner._run_action(
                     action_node=action_node, parent_execution_id=self.liveaction_id,
                     params=resolved_params)
             except Exception:
-                # Save the traceback and error message.
+                # Save the traceback and error message
                 LOG.exception('Failure in running action %s.', action_node.name)
                 error = {'error': traceback.format_exc(10)}
                 context_result[action_node.name] = error
+
+                if resolved_params is None:
```

Comment:
I disagree with this approach. Creating a LiveActionDB in a runner is an abstraction leak.